example/counterBasedRng/src/counterBasedRng.cpp

```diff
@@ -0,0 +1,258 @@
+
+    // Select a device
+    auto const devAcc = alpaka::getDevByIdx<Acc>(0u);
+    auto const devHost = alpaka::getDevByIdx<AccHost>(0u);
+
+    // Create a queue on the device
+    QueueAcc queueAcc(devAcc);
+    QueueHost queueHost(devHost);
+
+    // Define the work division
+    alpaka::Vec<Dim, Idx> const extent = {16, 16, 16 * 8};
+    alpaka::Vec<Dim, Idx> const elementsPerThread = {1, 1, 1};
+    alpaka::Vec<Dim, Idx> const elementsPerThreadHost = {1, 1, 8};
+
+    // Let alpaka calculate good block and grid sizes given our full problem extent
+    alpaka::WorkDivMembers<Dim, Idx> const workDivAcc(alpaka::getValidWorkDiv<Acc>(
+        devAcc,
+        extent,
+        elementsPerThread,
+        false,
+        alpaka::GridBlockExtentSubDivRestrictions::Unrestricted));
+    alpaka::WorkDivMembers<Dim, Idx> const workDivHost(alpaka::getValidWorkDiv<AccHost>(
+        devHost,
+        extent,
+        elementsPerThreadHost,
+        false,
+        alpaka::GridBlockExtentSubDivRestrictions::Unrestricted));
+
+    // Define the buffer element type
+    using Data = std::uint32_t;
+
+    // Allocate 3 host memory buffers
+    using BufHost = alpaka::Buf<DevHost, Data, Dim, Idx>;
+    BufHost bufHost(alpaka::allocBuf<Data, Idx>(devHost, extent));
+    BufHost bufHostDev(alpaka::allocBuf<Data, Idx>(devHost, extent));
+
+    // Initialize the host input vectors A and B
+    Data* const pBufHost(alpaka::getPtrNative(bufHost));
+    Data* const pBufHostDev(alpaka::getPtrNative(bufHostDev));
+
+    std::random_device rd{};
+    CounterBasedRngKernel::Key<AccHost> key = {rd(), rd()};
+
+    // Allocate buffer on the accelerator
+    using BufAcc = alpaka::Buf<Acc, Data, Dim, Idx>;
+    BufAcc bufAcc(alpaka::allocBuf<Data, Idx>(devAcc, extent));
+
+    // Create the kernel execution task.
+    auto const taskKernelAcc = alpaka::createTaskKernel<Acc>(
+        workDivAcc,
+        CounterBasedRngKernel(),
+        alpaka::experimental::writeAccess(bufAcc),
+        key);
+    auto const taskKernelHost = alpaka::createTaskKernel<AccHost>(
+        workDivHost,
+        CounterBasedRngKernel(),
+        alpaka::experimental::writeAccess(bufHost),
+        key);
+
+    // Enqueue the kernel execution task
+    alpaka::enqueue(queueHost, taskKernelHost);
+    alpaka::enqueue(queueAcc, taskKernelAcc);
+
+    // Copy the result from the device
+    alpaka::memcpy(queueAcc, bufHostDev, bufAcc);
+    const auto numElements = extent.prod();
+    // alpaka::memcpy(queueAcc,
+    //     alpaka::createView(devHost, pBufHostDev, alpaka::Vec<alpaka::DimInt<1u>, Idx>(numElements)),
+    //     alpaka::createView(devAcc, alpaka::getPtrNative(bufAcc),
+    //         alpaka::Vec<alpaka::DimInt<1u>, Idx>(numElements)));
+
+    // wait in case we are using an asynchronous queue to time actual kernel runtime
+    alpaka::wait(queueHost);
+    alpaka::wait(queueAcc);
+
+    int falseResults = 0;
+    const int MAX_PRINT_FALSE_RESULTS = extent[2] * 2;
```

Comment:
Please use camel or snake case for local variables.
```suggestion
    const int MAX_PRINT_FALSE_RESULTS = extent[2] * 2;
```